core/src/main/java/com/scalar/db/api/UpsertBuilder.java

```diff
@@ -105,6 +113,22 @@ public Buildable attributes(Map<String, String> attributes) {
       return this;
     }
 
+    @Override
+    public Buildable readTag(String policyName, String readTag) {
+      checkNotNull(policyName);
+      checkNotNull(readTag);
+      AbacOperationAttributes.setReadTag(attributes, policyName, readTag);
+      return this;
+    }
+
+    @Override
+    public Buildable writeTag(String policyName, String writeTag) {
```

Comment:
Just a confirmation. The reason for not having `writeTag` in InsertBuilder is that insertion uses the default tag?

Reply:
Actually, the `writeTag` is not the `rowTag`. The `writeTag` is used for access control on existing records. Since insert operations involve adding new records, the `writeTag` is not applicable to them. Therefore, the `InsertBuilder` does not include the `writeTag`. Thanks.